src/ToricVarieties/LineBundles/properties.jl

```diff
@@ -0,0 +1,87 @@
+@doc Markdown.doc"""
+    istrivial(l::ToricLineBundle)
+
+Returns if the toric line bundle `l` is trivial.
+
+# Examples
+```jldoctest
+julia> H = hirzebruch_surface(4)
+A normal, non-affine, smooth, projective, gorenstein, q-gorenstein, non-fano, 2-dimensional toric variety without torusfactor
+
+julia> istrivial(ToricLineBundle(H, [1,0]))
+false
+
+julia> istrivial(ToricLineBundle(H, [0,0]))
+true
+```
+"""
+function istrivial(l::ToricLineBundle)
+    return get_attribute!(l, :istrivial) do
+        return isprincipal(divisor(l))
+    end
+end
+export istrivial
+
+
+@doc Markdown.doc"""
+    isbasepoint_free(l::ToricLineBundle)
+
+Returns if the toric line bundle `l` is isbasepoint free.
+
+# Examples
+```jldoctest
+julia> H = hirzebruch_surface(4)
+A normal, non-affine, smooth, projective, gorenstein, q-gorenstein, non-fano, 2-dimensional toric variety without torusfactor
+
+julia> isbasepoint_free(ToricLineBundle(H, [1,0]))
+true
+```
+"""
+function isbasepoint_free(l::ToricLineBundle)
+    return get_attribute!(l, :isbasepoint_free) do
+        return isbasepoint_free(divisor(l))
```

Comment:
If the divisor caches this property (does it?), then do we really need to cache it here? I don't think it'll be substantially faster (though we could measure).

This would be different if we had "real" properties as in GAP, where (a) storing a property only takes 2 bits (one for its value, and one to know whether the value was set), and (b) there was a real "implication system" as in GAP. Perhaps we'll want to add such a thing at some point, but for now... ?

Reply:
Good point. I will adjust (and find out why all tests fail...)